app/scripts/controllers/newfromtemplate.js

```diff
@@ -176,7 +177,10 @@ angular.module('openshiftConsole')
 
         $scope.createFromTemplate = function() {
           $scope.disableInputs = true;
-          $scope.template.labels = keyValueEditorUtils.mapEntries(keyValueEditorUtils.compactEntries($scope.labels));
+          var userLabels = keyValueEditorUtils.mapEntries(keyValueEditorUtils.compactEntries($scope.labels));
+          var systemLabels = keyValueEditorUtils.mapEntries(keyValueEditorUtils.compactEntries($scope.systemLabels));
+          $scope.template.labels = _.extend(systemLabels, userLabels);
```

Comment:
This mutates `systemLabels`, but I don't think it's a problem since `mapEntries` and `compactEntries` probably create a copy.